### PR TITLE
Prepare for Bottlerocket v0.5.0 release

### PR DESCRIPTION
**Pull request by etungsten**
**Issue number:**
N/A

**Description of changes:**
Prepares for Bottlerocket v0.5.0 release
- Adds v0.5.0 changelog
- Updates `Release.toml`

**Testing done:**
N/A

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,71 @@
...
+## OS changes
+
+* Add `aws-ecs-1` variant prototype for running containers in ECS clusters ([#946])
+* Configurable `clusterDomain` kubelet setting via `settings.kubernetes.cluster-domain` ([#988])
+* Make update position within waves consistent ([#993])
+* Fix kubelet configuration for `MaxPods` ([#994])
+* Update `eni-max-pods` with new instance types ([#994])
+* Fix `max_versions` unit test in `updata` ([#998])
+* Add configurable logging drivers for `aws-ecs-1` variant for use with ECS ([#1005])
+* Add IAM task role capabilities support for `aws-ecs-1` variant ([#1007])
```
Comment (webern) [inline CHANGELOG.md]: Don't all of these get listed as 'added a new variant for ECS`... #1005, #10007, etc

**Review by webern — APPROVED** [review]

**Comment by etungsten** [comment]
Push above addresses @webern 's comment.

Merged all items related the to the `aws-ecs-1` variant into the first `aws-ecs-1` line item.

**Comment by etungsten** [comment]
Rebases on top of develop.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,66 @@
...
+
+## OS changes
+
+* Add `aws-ecs-1` variant prototype for running containers in ECS clusters ([#946], [#1005], [#1007], [#1008], [#1009], [#1017])
+* Configurable `clusterDomain` kubelet setting via `settings.kubernetes.cluster-domain` ([#988])
+* Make update position within waves consistent ([#993])
+* Fix kubelet configuration for `MaxPods` ([#994])
+* Update `eni-max-pods` with new instance types ([#994])
+* Fix `max_versions` unit test in `updata` ([#998])
+* Remove injection of `label:disable` option for privileged containers in Docker ([#1013])
```
Comment (tjkirch) [inline CHANGELOG.md:17]: Also up with the ecs item?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,66 @@
...
+* Add `policycoreutils` and related tools ([#1016])
+* Update third-party software packages ([#1018])
+* Update Rust dependencies ([#1019])
+* Update `host-ctr`'s dependencies ([#1020])
+* Fix unit test in `migrator` to use new `tough` 0.8 interface ([#1021])
```
Comment (tjkirch) [inline CHANGELOG.md]: This can go with 1019.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,66 @@
...
+* Fix warnings for use of deprecated items in `common_migrations` ([#1022])
```
Comment (tjkirch) [inline CHANGELOG.md]: This can go in Build changes, I think, since it only affects the build's CLI output.

File: CHANGELOG.md (hunk 2 above)
Comment (etungsten) [reply CHANGELOG.md:17]: This also applies to `aws-dev` variant since it also uses docker.

**Comment by etungsten** [comment]
Push above rebases onto develop

**Comment by etungsten** [comment]
Addresses @tjkirch 's comments.
Adds #1010 to the list of changes.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,66 @@
...
+* Remove injection of `label:disable` option for privileged containers in Docker ([#1013])
+* Add `policycoreutils` and related tools ([#1016])
+* Update third-party software packages ([#1018])
+* Update Rust dependencies ([#1019], [#1021])
+* Update `host-ctr`'s dependencies ([#1020])
+* Update SElinux packages to v3.1 ([#1023])
+
+## Build changes
+
+* Add `pubsys` (`cargo make repo`) for repo creation/update and AMI registration/copying ([#964], [#1010])
```
Comment (tjkirch) [inline CHANGELOG.md]:
```suggestion
* Add `pubsys` (`cargo make repo`, `cargo make ami`) for repo and AMI creation ([#964], [#1010])
```

**Review by tjkirch — APPROVED** [review]

**Comment by etungsten** [comment]
Push above addresses https://github.com/bottlerocket-os/bottlerocket/pull/1024#discussion_r468245680

**Comment by etungsten** [comment]
Push above rebases on develop to add in the last changes needed for v0.5.0

**Review by samuelkarp — APPROVED** [review]

**Review by tjkirch — APPROVED** [review]

**Pull request merged**